go.mod

```diff
@@ -1,60 +1,74 @@
 module sigs.k8s.io/apiserver-network-proxy
 
-go 1.17
+go 1.18
 
 require (
-	github.com/golang/mock v1.4.4
-	github.com/golang/protobuf v1.4.3
-	github.com/google/uuid v1.1.2
-	github.com/prometheus/client_golang v1.7.1
-	github.com/spf13/cobra v0.0.3
+	github.com/golang/mock v1.6.0
+	github.com/golang/protobuf v1.5.2
+	github.com/google/uuid v1.3.0
+	github.com/prometheus/client_golang v1.13.0
+	github.com/spf13/cobra v1.5.0
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/net v0.0.0-20201110031124-69a78807bb2b
-	google.golang.org/grpc v1.42.0
-	k8s.io/api v0.20.10
-	k8s.io/apimachinery v0.20.10
-	k8s.io/client-go v0.20.10
-	k8s.io/component-base v0.20.10
-	k8s.io/klog/v2 v2.4.0
+	golang.org/x/net v0.0.0-20220812174116-3211cb980234
+	google.golang.org/grpc v1.48.0
+	google.golang.org/protobuf v1.28.1
+	k8s.io/api v0.24.3
+	k8s.io/apimachinery v0.24.3
+	k8s.io/client-go v0.24.3
+	k8s.io/component-base v0.24.3
+	k8s.io/klog/v2 v2.70.1
 	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.0
 )
 
 require (
```

Comment:
this added a *lot* of new dependencies, many of which we explicitly don't want to depend on transitively in Kubernetes:

* github.com/go-openapi/...
* new incompatible version of github.com/imdario/mergo (xref https://github.com/kubernetes/kubernetes/issues/107499)

Also, this only updated the root go.mod file, not the client konnectivity-client/go.mod file, so now this repo has two disparate dependency trees. If the client go.mod file was updated the same way this was, we would be unable to pull new versions into kubernetes.

How were the versions of these updated modules selected?

Reply:
The two that you mentioned are both transitive dependencies through client-go. I don't know how that specific client-go version was selected, but since this PR sat open for a long time I'm guessing it was the latest version when this PR was opened.

We haven't been keeping the konnectivity-client go.mod in sync with this since we're tracking the module versions from the oldest version of k8s we support in the client. We should maybe consider adding a presubmit to ensure that konnectivity-client doesn't take any dependencies on kubernetes staging repos, but hopefully our PR review process would catch that.

https://github.com/kubernetes/client-go/blob/v0.24.3/go.mod#L26

![openapi](https://user-images.githubusercontent.com/29742491/206006710-45ffc614-44f5-496b-94f8-ce68d22bd510.png)

Reply:
oops, I misread the go-openapi additions, these are the ones we want to avoid:
```
      "github.com/go-openapi/analysis": "use k8s.io/kube-openapi/pkg/validation/spec",
      "github.com/go-openapi/spec": "use k8s.io/kube-openapi/pkg/validation/spec instead",
      "github.com/go-openapi/strfmt": "use k8s.io/kube-openapi/pkg/validation/strfmt instead",
      "github.com/go-openapi/validate": "use k8s.io/kube-openapi/pkg/validation/validate instead",
```

the github.com/imdario/mergo one is a known dependency but the version selected here is known to be problematic (see the linked issue above)